__tests__/app.test.js

```diff
@@ -38,4 +39,35 @@ describe("GET/topics", () => {
             expect(topicsArray[2].description).toBe('what books are made of')
         })
     })
+})
+
+describe("GET/api", () => {
+    test("200: when calling the api as the endpoint, return an object containing all of the endpoint information for every endpoint tested in app.js", () => {
+        return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body }) => {
+            const { endpointObject } = body
+            return fs.readFile('./endpoints.json', 'utf-8')
+                .then((fileContents) => {
+                    const parsedFile = JSON.parse(fileContents)
+                    expect(endpointObject).toEqual(parsedFile)
+                })
+        })
+    })
+    test("200: For each endpoint, make sure that all of the required information and keys are there", () => {
+        return request(app)
+            .get("/api")
+            .expect(200)
+            .then(({ body }) => {
+                const { endpointObject } = body;
+                expect(endpointObject["GET /api/topics"].exampleResponse).not.toBeUndefined();
+                expect(endpointObject["GET /api/topics"].description).not.toBeUndefined();
+                expect(endpointObject["GET /api/topics"].queries).not.toBeUndefined();
+    
+                expect(endpointObject["GET /api/articles"].description).not.toBeUndefined();
+                expect(endpointObject["GET /api/articles"].queries).not.toBeUndefined();
+                expect(endpointObject["GET /api/articles"].exampleResponse).not.toBeUndefined();
+            });
```

Comment:
This second test is entirely covered by the first test above, so it's not necessary to include, as the first test uses `toEqual` it will already be checking that these properties are there